flang/lib/Lower/ReductionRuntime.cpp

```diff
@@ -86,6 +86,121 @@ struct ForcedMinvalInteger16 {
   }
 };
 
+/// Placeholder for real*10 version of Product Intrinsic
+struct ForcedProductReal10 {
+  static constexpr const char *name = ExpandAndQuoteKey(RTNAME(ProductReal10));
+  static constexpr Fortran::lower::FuncTypeBuilderFunc getTypeModel() {
+    return [](mlir::MLIRContext *ctx) {
+      auto ty = mlir::FloatType::getF80(ctx);
```

Comment:
You should use the same `getModel` code to construct the signatures as done in the reflection cases whenever possible. This case, `getF80`, is the exception.